lib/src/components/NxRadio/__tests__/NxRadio.test.tsx

```diff
@@ -21,146 +22,131 @@ describe('NxRadio', function() {
-      expect(getShallow({ disabled: true })).toHaveClassName('nx-radio-checkbox--disabled');
+      expect(renderEl()).not.toHaveClass('nx-radio-checkbox--disabled');
+      expect(renderEl({ disabled: true })).toHaveClass('nx-radio-checkbox--disabled');
     });
 
     it('disables the input', function() {
-      expect(getShallow().find('input')).toHaveProp('disabled', false);
-      expect(getShallow({ disabled: false }).find('input')).toHaveProp('disabled', false);
-      expect(getShallow({ disabled: true }).find('input')).toHaveProp('disabled', true);
+      expect(renderEl()?.querySelector('input')).not.toHaveAttribute('disabled');
+      expect(renderEl({ disabled: false })?.querySelector('input')).not.toHaveAttribute('disabled');
+      expect(renderEl({ disabled: true })?.querySelector('input')).toHaveAttribute('disabled');
     });
   });
 
   describe('isChecked prop', function () {
     it('sets the input to checked per the value of isChecked', function() {
-      expect(getShallow({ isChecked: false }).find('input')).toHaveProp('checked', false);
-      expect(getShallow({ isChecked: true }).find('input')).toHaveProp('checked', true);
+      expect(renderEl({ isChecked: false })?.querySelector('input')).not.toHaveAttribute('checked');
+      expect(renderEl({ isChecked: true })?.querySelector('input')).toHaveAttribute('checked');
     });
 
     it('adds the tm-checked class if isChecked is true, and the tm-unchecked class if it is false', function() {
-      expect(getShallow()).toHaveClassName('tm-unchecked');
-      expect(getShallow()).not.toHaveClassName('tm-checked');
+      expect(renderEl()).toHaveClass('tm-unchecked');
+      expect(renderEl()).not.toHaveClass('tm-checked');
 
-      expect(getShallow({ isChecked: true })).not.toHaveClassName('tm-unchecked');
-      expect(getShallow({ isChecked: true })).toHaveClassName('tm-checked');
+      expect(renderEl({ isChecked: true })).not.toHaveClass('tm-unchecked');
+      expect(renderEl({ isChecked: true })).toHaveClass('tm-checked');
     });
   });
 
   describe('children prop', function () {
     it('renders children nodes within an nx-radio__content <span>', function() {
-      const render = shallow(
-        <NxRadio { ...simpleProps }>
+      const { container, rerender } = quickRender();
+
+      rerender(
+        <NxRadio { ...minimalProps }>
           <div className="bar"></div>
         </NxRadio>
       );
 
-      expect(render).toContainMatchingElement('label span.nx-radio__content .bar');
+      expect(container.querySelector('span.nx-radio__content .bar')).toBe;
     });
 
     it('does not render the .nx-radio__content element if there are no children', function() {
-      expect(getShallow()).not.toContainMatchingElement('.nx-radio__content');
+      const el = renderEl()!;
+      expect(el).not.toHaveClass('nx-radio__content');
     });
-
-    it('wraps the .nx-radio__content element in an NxOverflowTooltip unless overflowTooltip is set to false',
```

Comment:
Added a visual test for the tooltip. Andrew and I were talking about it, and it was getting complicated to convert to RTL so we thought a functional test would be better.

Reply:
> > Added a visual test for the tooltip. Andrew and I were talking about it, and it was getting complicated to convert to RTL so we thought a functional test would be better.
> 
> Hmm, the visual test doesn't cover the `overflowTooltip` prop though, does it? It's true that `NxOverflowTooltip` can't meaningfully be tested in unit tests though, because the DOM layout logic that it relies on is not implemented in JSDOM. So testing it in functional tests is the right way to go, but I suspect the current test doesn't fully test the prop.

What would be a good way to test it, can I maybe add an expect there to check if the radio has NxOverflowTooltip as an attribute or would that not work?

Reply:
> What would be a good way to test it, can I maybe add an expect there to check if the radio has NxOverflowTooltip as an attribute or would that not work?

In my experience the `userEvent.hover` function does work for triggering tooltips in unit tests.  You can use that on an element and then do `screen.getByRole('tooltip')` to detect whether the tooltip appeared.

Reply:
I added the visual tests for the Radio label does not wrap example here: ec817cb345ce1fc17da4ea313ca1bf66bc744166